### PR TITLE
ept_test: New case to test when ept is off

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1816910
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.ept_test: PASS (70.96 s)

**Comment by huangyum** [comment]
@luckyh @yanan-fu Please help review, thanks.

File: qemu/tests/ept_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    if cpu.get_cpu_vendor_name() != 'intel':
+        test.cancel("This test is supposed to run on Intel host")
+
+    unload_cmd = params["unload_cmd"]
+    load_cmd = params["load_cmd"]
+    test_cmd = params["test_cmd"]
+    process.system(unload_cmd)
+    process.system(load_cmd % "0")
+    process.system(test_cmd)
```
Comment (yanan-fu) [inline qemu/tests/ept_test.py]: @huangyum  Only `cat` ?  why do not check the content of `vmentry_l1d_flush` ? `EPT disabled` when `ept=0`

Comment (huangyum) [reply qemu/tests/ept_test.py]: Yes, they are the steps to trigger a system crash before. You can refer to the bug linked to polarion case. And the content may vary on different hosts.

Comment (yanan-fu) [reply qemu/tests/ept_test.py]: Got it, thanks!

File: qemu/tests/ept_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from avocado.utils import process, cpu
+from virttest import env_process
+
+
+def run(test, params, env):
+    """
+    ept test:
+    1) Turn off ept on host
+    2) Check kvm_intel parameter
```
Comment (yanan-fu) [inline qemu/tests/ept_test.py]: could you make this more clear ? the purpose is check if read param cause host crash as we discussed.

Comment (huangyum) [reply qemu/tests/ept_test.py]: Updated.

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

File: qemu/tests/ept_test.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    """
+
+    if cpu.get_cpu_vendor_name() != 'intel':
+        test.cancel("This test is supposed to run on Intel host")
+
+    unload_cmd = params["unload_cmd"]
+    load_cmd = params["load_cmd"]
+    test_cmd = params["test_cmd"]
+    process.system(unload_cmd)
+    process.system(load_cmd % "0")
```
Comment (luckyh) [inline qemu/tests/ept_test.py]: Should record the value of the ept option before doing the test and restore at the end of the test.

Comment (huangyum) [reply qemu/tests/ept_test.py:27]: Thanks, updated.

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.ept_test: PASS (71.12 s)

File: qemu/tests/ept_test.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    :param env: Dictionary with test environment.
+    """
+
+    if cpu.get_cpu_vendor_name() != 'intel':
+        test.cancel("This test is supposed to run on Intel host")
+
+    unload_cmd = params["unload_cmd"]
+    load_cmd = params["load_cmd"]
+    read_cmd = params["read_cmd"]
+    ept_value = process.system_output(read_cmd % "ept")
```
Comment (luckyh) [inline qemu/tests/ept_test.py:25]: To ensure it performs the operation properly, the logic should be:
```python
ept_value = process.system_output(read_cmd % "ept")
try:
    # run the test
    ...
finally:
    # restore the value
    ...
```

**Comment by huangyum** [comment]
@luckyh Any more comments?

**Pull request merged**